mango/shared/spatial.py

```diff
@@ -0,0 +1,30 @@
+from math import radians, cos, sin, asin, sqrt
+from .decorators import pydantic_validation
+from pydantic import BaseModel, confloat
+from typing import Tuple
+
+class HaversineArgs(BaseModel):
+    point1: Tuple[confloat(ge=-90.0, le=90.0),confloat(ge=-180.0, le=180.0)]
+    point2: Tuple[confloat(ge=-90.0, le=90.0),confloat(ge=-180.0, le=180.0)]
+
```

Comment:
I'm not a 100% sure that these clases have to be defined in the same file as the function in which they are used or in another section of the library so that the can (maybe) be used in other parts of mango or outside of mango